docs/_includes/sidebar.html

```diff
@@ -1,46 +1,45 @@
-{% comment %}
-    Create an empty array to store secondary nav sections
-{% endcomment %}
-{% assign secondary_sections = '' | split: ',' %}
-
-{% for p in site.pages %}
-    {% if p.section == page.section and p.secondary_section %}
-        {% unless secondary_sections contains p.secondary_section %}
-            {% assign secondary_sections = secondary_sections | push: p.secondary_section %}
-        {% endunless %}
-    {% endif %}
-{% endfor %}
-
-{% comment %}
-    Alphabetize the sections
-{% endcomment %}
-{% assign secondary_sections = secondary_sections | sort %}
-
-{% if secondary_sections %}
 <ul class="m-list m-list__unstyled">
-    {% for secondary_section in secondary_sections | sort %}
+{% for first in site.data.side-navigation.first-level %}
+    {% assign first_level_slug = first.heading | slugify %}
     <li class="m-list_item u-mb15">
-        <span class="h5">{{ secondary_section }}</span>
-        <ul class="m-list m-list__links">
-            {% for p in site.pages %}
-                {% if p.section == page.section and p.secondary_section == secondary_section %}
+        <div class="h4">
+            <a class="m-list_link" href="{{ first_level_slug | relative_url }}">{{ first.heading }}</a>
+        </div>
+        {% if page.section == first_level_slug or page == blank %}
+        <div class="m-list_item__tertiary">
```

Comment:
Modifier should be used with a block or element to modify it. So this should be looking like 

```
m-list_item m-list_item__tertiary
```

…where `m-list_item` is the element that is adding some styles, and `m-list_item__tertiary` is the modifier changing some styles of that element.

This doesn't appear to me that you need to necessarily hook into the `m-list` architecture at all. If you did, this would be an element, like `m-list_item-subgroup` or similar, but if you didn't you could make this a utility class, `u-ml20` (for margin-left 20px, to fit with our existing utilities like this).